packages/openapi-code-generator/src/typescript/common/model-builder.ts

```diff
@@ -132,20 +132,21 @@ export class ModelBuilder {
         const members = Object.entries(schemaObject.properties)
           .sort(([a], [b]) => a < b ? -1 : 1)
           .map(([name, definition]) => {
+            const isRequired = schemaObject.required.some(it => it === name)
 
             if (isRef(definition)) {
               this.add(definition)
 
               return [
                 `"${name}"`,
+                isRequired ? "" : "?",
```

Comment:
This is the fix, the rest is just cleaning up the raw output a bit to make the unit test snapshot look nicer